library/dokku_http_auth.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+import subprocess
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.dokku_utils import subprocess_check_output
+
+DOCUMENTATION = """
+---
+module: dokku_http_auth
+short_description: Manage HTTP Basic Authentication for a dokku app
+options:
+  app:
+    description:
+      - The name of the app
+    required: True
+    default: null
+    aliases: []
+  state:
+    description:
+      - The state of the http-auth plugin
+    required: False
+    default: present
+    choices: [ "present", "absent" ]
+    aliases: []
+  username:
+    description:
+      - The HTTP Auth Username (required for 'present' state)
+    required: False
+    aliases: []
+  password:
+    description:
+      - The HTTP Auth Password (required for 'present' state)
+    required: False
+    aliases: []
+author: Simo Aleksandrov
+requirements:
+  - the `dokku-http-auth` plugin
+"""
+
+EXAMPLES = """
+- name: Enable the http-auth plugin
+  dokku_http_auth:
+    app: hello-world
+    state: present
+    username: samsepi0l
+    password: hunter2
+
+- name: Disable the http-auth plugin
+  dokku_http_auth:
+    app: hello-world
+    state: absent
+"""
+
+
+def dokku_http_auth_enabled(data):
+    command = "dokku --quiet http-auth:report {0}"
+    response, error = subprocess_check_output(command.format(data["app"]))
+
+    if error:
+        return None, error
+
+    report = response[0].split(":")[1]
+    return report.strip() == "true", error
+
+
+def dokku_http_auth_present(data):
+    is_error = True
+    has_changed = False
+    meta = {"present": False}
+
+    enabled, error = dokku_http_auth_enabled(data)
```

Comment:
What if an error is returned here?
It seems to me this is currently silently ignored...

Reply:
oops, must have forgotten this